core/src/main/scala/org/http4s/headers/Accept-Charset.scala

```diff
@@ -14,7 +14,7 @@ final case class `Accept-Charset`(values: NonEmptyList[CharsetRange]) extends He
   type Value = CharsetRange
 
   def qValue(charset: Charset): QValue = {
-    def specific = values.collectFirst { case cs: CharsetRange.Atom => cs.qValue }
+    def specific = values.collectFirst { case cs: CharsetRange.Atom if cs.isSatisfiedBy(charset) => cs.qValue }
     def splatted = values.collectFirst { case cs: CharsetRange.`*` => cs.qValue }
```

Comment:
Just checking as the PR also had this line modified if this was the intention?

Reply:
`isSatisfiedBy` was wrong, because it would let zeroes slide through and fall back to a splat.  I introduced `matches` in 178d075.

Splatted was skipped because it matches all charsets.  We could make it explicit, but it would essentially be a method call to `if true`.